Add MGSmootherPrecondition::initialize_matrices()

peterrum opened the pull request:
I would like to setup `MGSmootherPrecondition` level by level, not using the `initialize()` functions, which also initialize the preconditioners. The best would be to have a function `initialize(const MGLevelObject<MatrixType2> &matrices)` but that conflicts with the function with the same name that takes optionally an `AdditonalData` object.

kronbichler commented on include/deal.II/multigrid/mg_smoother.h:
```diff
@@ -467,6 +467,15 @@ class MGSmootherPrecondition : public MGSmoother<VectorType>
 
+  /**
+   * In contrast to the function above, only initialize the matrices. The
+   * smoothers need to be setup manually by the user in the following. This
```
```suggestion
   * smoothers need to be set up manually by the user as a subsequent step in the code. This
```

kronbichler commented on include/deal.II/multigrid/mg_smoother.h:
```diff
@@ -467,6 +467,15 @@ class MGSmootherPrecondition : public MGSmoother<VectorType>
+  /**
+   * In contrast to the function above, only initialize the matrices. The
+   * smoothers need to be setup manually by the user in the following. This
+   * is useful if one wants full flexibility in the choice of smoothers.
```
Maybe you can explain a bit more how this works: By manually setting the public member variable `smoothers`?

kronbichler reviewed (approved):
OK with me, but I must say that I am surprised that we expose the smoothers as public member variables in this class, which makes this way of initialization possible at all. But I guess it is as you say, one might want to have additional flexibility in the solver.

peterrum commented:
> OK with me, but I must say that I am surprised that we expose the smoothers as public member variables in this class, which makes this way of initialization possible at all.

I was also surprised by this. I think a setter function allowing controlled access to the level smoothers would have been a better choice.

@kronbichler I have addressed your comments!

kronbichler commented:
/rebuild
